### DISCUSSION TITLE
How do I clear a page of annotations/drawings with canvas mode?

### DISCUSSION THREAD

**Discussion by usagibear** [wojtekmaj/react-pdf#1416] [Q&A]
I am trying to draw and mark up the page canvas. I need to implement drawing, panning, undo, erase, and clear.

When I am panning and clearing, I am trying to manipulate the underlying page with translations but the page itself does not move. When I draw new shapes, they are drawn with the translation. Is this underlying page (the PDF page) special some how?

When I try to clear with `clearRect`, it wipes out the drawing itself and overlays white. Is there a way for me to re-render the page? I saw that there is a method called `drawPageOnCanvas` in `PageCanvas.jsx` but I do not know how to call it, if I can, or if there is a better way.

I am still new to using Canvas and React so do not know if this it the right approach for trying to annotate or manipulate the PDF page. Thanks for the help!

**Comment by wojtekmaj** [comment]
You could pass a different `key` prop to `Page` component to force rerender it.
